### PR TITLE
Including all the gpu versions mentioned in the backends attribute.

### PR DESCRIPTION
**Pull request by hmonishN**
Added all the gpu version mentioned in the backends field in BUILD files. The tests marked with gpu_v100, gpu_a100, gpu_h100, etc were not getting picked up by bazel during execution. The fix will include the tests and run them on respective gpu nodes.

### CODE REVIEW

**Comment by hmonishN** [comment]
+ @ddunleavy@google.com

File: xla/tests/build_defs.bzl [hunk 1]
```diff
@@ -111,7 +111,7 @@ def xla_test(
         if backend == "cpu":
             backend_deps = ["//xla/service:cpu_plugin"]
             backend_deps += ["//xla/tests:test_macros_cpu"]  # buildifier: disable=list-append
-        elif backend == "gpu":
+        elif backend == "gpu" or backend == "gpu_a100" or backend == "gpu_v100" or backend == "gpu_p100" or backend == "gpu_h100":
```
Comment (ddunl) [inline xla/tests/build_defs.bzl:114]: I think it'd be nice to have a `gpu_backends` list at the top with `all_backends`, and then redefine like `all_backends = gpu_backends + ["cpu"] + plugins.keys()`

Comment (hmonishN) [reply xla/tests/build_defs.bzl]: Incorporated review comments.

**Comment by ddunl** [comment]
In parallel here I'm going to change the meaning of `tf_cuda_tests_tags` because I think this PR is exposing some inconsistencies in how we are using backends in `xla_test`, and also the associated tags for indicating different hardware. Another note - I think that we want to rename `gpu` -> `gpu_any` in this scheme to indicate when we don't care about the specific hardware used, just that we want a GPU. Sorry for the back and forth here!

**Comment by ddunl** [comment]
Closing now that https://github.com/openxla/xla/commit/c0f10b31c05590250d6a02c4911d84d3937bf904 is in. Thanks for this PR!!

**Pull request closed without merging**